File: beecrowd/Python/iniciante/1049.py

vertebrado = {
    "ave": {
        "carnivoro": "aguia",
        "onivoro": "pomba"
    },

    "mamifero": {
        "onivoro": "homem",
        "herbivoro": "vaca"
    }
}

invertebrado = {
    "inseto": {
        "hematofago": "pulga",
        "herbivoro": "lagarta"
    },
    "anelideo": {
        "hematofago": "sanguessuga",
        "onivoro": "minhoca"
    }
}

def animal(input1: str,input2:str,input3:str):
    if input1 == "vertebrado":
        print(vertebrado[f"{input2}"][f"{input3}"])
    elif input1 == "invertebrado":
        print(invertebrado[f"{input2}"][f"{input3}"])
    else:
        print("Entrada invalida")

if __name__ == "__main__":
    n1 = str(input())
    n2 = str(input())
    n3 = str(input())
    animal(n1,n2,n3)
    
    

File: beecrowd/Cpp/iniciante/1049.cpp
#include <iostream>
#include <map>
#include <string>

using namespace std;

int main(){

    map<string, map<string, map<string, string>>> animal;
    string n1,n2,n3;

    animal["vertebrado"]["ave"]["carnivoro"] = "aguia";
    animal["vertebrado"]["ave"]["onivoro"] = "pomba";
    animal["vertebrado"]["mamifero"]["onivoro"] = "homem";
    animal["vertebrado"]["mamifero"]["herbivoro"] = "vaca";

    animal["invertebrado"]["inseto"]["hematofago"] = "pulga";
    animal["invertebrado"]["inseto"]["herbivoro"] = "lagarta";
    animal["invertebrado"]["anelideo"]["hematofago"] = "sanguessuga";
    animal["invertebrado"]["anelideo"]["onivoro"] = "minhoca";

    cin >> n1;
    cin >> n2;
    cin >> n3;

    cout << animal[n1][n2][n3] << endl;

    return 0;
}